cpp/src/io/json/read_json.cu

```diff
@@ -232,13 +243,10 @@ datasource::owning_buffer<rmm::device_buffer> get_record_range_raw_input(
   auto const delimiter                = reader_opts.get_delimiter();
   auto const num_extra_delimiters     = num_delimiter_chars * sources.size();
   std::size_t const chunk_offset      = reader_opts.get_byte_range_offset();
-  std::size_t chunk_size              = reader_opts.get_byte_range_size();
-
-  CUDF_EXPECTS(total_source_size ? chunk_offset < total_source_size : !chunk_offset,
-               "Invalid offsetting",
-               std::invalid_argument);
-  auto should_load_till_last_source = !chunk_size || chunk_size >= total_source_size - chunk_offset;
-  chunk_size = should_load_till_last_source ? total_source_size - chunk_offset : chunk_size;
+  std::size_t const chunk_size        = reader_opts.get_byte_range_size();
+  // Sanity checks for the byte range offset and size are handled by the batching logic.
+  // We only need to check if we are reading until the end of the last source in this function.
+  auto const should_load_till_last_source = chunk_offset + chunk_size == total_source_size;
```

Comment:
It could be a stylistic preference, but did you intend for this variable to be of type `std::size_t` instead of `bool`?